File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/ApplicationInjector.java
package com.savvasdalkitsis.gameframe.injector;

import com.savvasdalkitsis.gameframe.GameFrameApplication;

public class ApplicationInjector {

    private static GameFrameApplication gameFrameApplication;

    public static void injectApplication(GameFrameApplication gameFrameApplication) {
        ApplicationInjector.gameFrameApplication = gameFrameApplication;
    }

    public static GameFrameApplication application() {
        return gameFrameApplication;
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/ApplicationInjector.kt
package com.savvasdalkitsis.gameframe.injector

import com.savvasdalkitsis.gameframe.GameFrameApplication

object ApplicationInjector {

    private var gameFrameApplication: GameFrameApplication? = null

    fun injectApplication(gameFrameApplication: GameFrameApplication) {
        ApplicationInjector.gameFrameApplication = gameFrameApplication
    }

    fun application(): GameFrameApplication = gameFrameApplication!!
}
